livingpark_utils/download/ppmi.py

```diff
@@ -0,0 +1,270 @@
+        for _, row in cohort.iterrows():
+            if not row["File name"] or row["File name"] is None:
+                filename = ppmi_fd.find_nifti(
+                    row["PATNO"], row["EVENT_ID"], row["Description"]
+                )
+                if filename is None:
+                    print(
+                        "Not found: "
+                        + f"{row['PATNO'], row['EVENT_ID'], row['Description']}"
+                    )
+                else:  # copy file to dataset
+                    dest_dir = os.path.join(
+                        "inputs",
+                        f'sub-{row["PATNO"]}',
+                        f'ses-{row["EVENT_ID"]}',
+                        "anat",
+                    )
+                    os.makedirs(dest_dir, exist_ok=True)
+                    dest_file = os.path.join(dest_dir, os.path.basename(filename))
+                    os.rename(filename, dest_file)
+                    row["File name"] = dest_file
+
+        # Update file names in cohort
+        cohort["File name"] = cohort.apply(
+            lambda x: ppmi.find_nifti_file_in_cache(
+                x["PATNO"], x["EVENT_ID"], x["Description"]
+            ),
+            axis=1,
+        )
+
+        # Create symlinks to inputs if necessary
+        if symlink:
+            for file_name in cohort["File name"].values:
+                dest_dir = os.path.dirname(file_name).replace(
+                    os.path.join(self.cache_dir, "inputs"),
+                    os.path.join(results_path, "pre_processing"),
+                )
+                dest_file = os.path.join(
+                    dest_dir,
+                    os.path.basename(file_name.replace(self.cache_dir, "")),
+                )
+                if not os.path.exists(dest_file):
+                    os.makedirs(dest_dir, exist_ok=True)
+                    os.symlink(
+                        os.path.relpath(os.path.abspath(file_name), start=dest_file),
+                        dest_file,
+                    )
+
+        return query, pd.DataFrame()
+
+    def missing_raw_data(
+        self, query: pd.DataFrame, *, force: bool = False
+    ) -> pd.DataFrame:
+        """Determine the missing raw data locally.
+
+        Parameters
+        ----------
+        query : pd.DataFrame
+            Cohort to verify.
+        force : bool, optional
+            When `True`, all metadata are reported missing locally., by default False
+
+        Returns
+        -------
+        pd.DataFrame
+            Missing raw data locally.
+        """
+        if force:
+            return query
+        else:
+            cohort = query
+            cohort["File name"] = cohort.apply(
+                lambda x: ppmi.find_nifti_file_in_cache(
+                    x["PATNO"], x["EVENT_ID"], x["Description"]
+                ),
+                axis=1,
+            )
+            return cohort[cohort["File name"] == ""]
+
+    def get_derivative(
```

Comment:
not sure what "the derivative" means, there can be as many derivatives as pipeline and parametrizations

Reply:
My initial idea for the "derivative" was for downloading preprocessed data to avoid computation. However,  I'm not sure if it makes sense to have this type of feature. If so, a different name could help to avoid confusion.

Reply:
Shall we add it only when the needs come? I don't know if we will need that, if we will need a "download paper pre-processed data" or nothing at all. If we do something, this function will most likely require an id of the paper where the analysis is defined.